src/UnitOfWork.php

```diff
@@ -2390,29 +2390,27 @@ private function doDetach(
 
         $visited[$oid] = $entity; // mark visited
 
-        switch ($this->getEntityState($entity, self::STATE_DETACHED)) {
-            case self::STATE_MANAGED:
-                if ($this->isInIdentityMap($entity)) {
-                    $this->removeFromIdentityMap($entity);
-                }
+        $state = $this->getEntityState($entity, self::STATE_DETACHED);
```

Comment:
I had to make this changes because "child" objects should be detached first, and then the parents (otherwise the IDs are not available for some DBs as sqlite)

Reply:
I am afraid this is going to break some people event listeners :) 

We should add this the UPGRADE notes to warn people. We don't guarantee the order so its not a BC break, but a heads up is always nice.